raspberry/bzzz/sensors/gps.py

```diff
@@ -0,0 +1,98 @@
+import serial
+import time
+from datetime import datetime, timedelta
+from threading import Thread, Lock
+from data_logger import DataLogger  # Ensure this is correctly imported or defined in the same file
+
+def deg_min_sec_to_decimal(degrees, minutes, direction):
+    """
+    Convert degrees-minutes-seconds from GNGLL to decimal
+    :param degrees: degrees
+    :param minutes: minutes
+    :param direction: direction (can be one of 'N', 'W', 'E', 'S')
+    :return: decimal representation of GPS coordinates
+    """
+    decimal_degrees = degrees + minutes / 60
+    if direction in ['S', 'W']:  # need to be denoted with a negative value
+        decimal_degrees *= -1
+    return decimal_degrees
+
+class Gps:
+    """
+    Initializes the GPS data handler with the default serial path and baud rate,
+    sets up storage for latitude, longitude, and altitude,
+    and starts a background thread to continuously read and parse GPS data.
+    :param serial_path: Path to the serial port where the GPS is connected (default: "/dev/ttyACM0")
+    :param baud: Baud rate for the serial connection (default: 500000)
+    :param log_file: Name of the file to log GPS data (default: "gps_data.csv")
+    """
+    def __init__(self, serial_path="/dev/ttyACM0", baud=500000, log_file="gps_data.csv"):
+        self.__lock = Lock()
+        self.__gpgll_latitude = None
+        self.__gpgll_longitude = None
+        self.__gpgsv_altitude = None
+        # Initialize DataLogger with 3 features: Timestamp, Latitude, Longitude, Altitude
+        self.logger = DataLogger(num_features=4, feature_names=["Timestamp", "Latitude", "Longitude", "Altitude"])
+        self.__thread = Thread(target=self.__get_measurements_in_background, args=[serial_path, baud])
+        self.__keep_going = True
+        self.__thread.start()
+        self.log_file = log_file
+
+    def __get_measurements_in_background(self, serial_path, baud):
+        """
+        Continuously reads GPS data from the serial port, parsing GNGLL messages
+        for latitude and longitude, and GPGSV messages for altitude,
+        updating the object's GPS attributes accordingly. This method will write
+        the GPS data to a CSV file for 2 minutes.
+        :param serial_path: Serial port path
+        :param baud: Baud rate for the serial connection
+        """
+        stop_time = datetime.now() + timedelta(minutes=2)  # Determine when to stop collecting data
```

Comment:
Should we not be continuously be reading GNSS data in background so don't think it stop after 2 minutes

Reply:
@jamie-54 since we're in a bit of a hurry now, can you please take care of this PR?